module/zfs/zio_compress.c

```diff
@@ -82,6 +100,8 @@ zio_compress_select(spa_t *spa, enum zio_compress child,
 	if (result == ZIO_COMPRESS_ON) {
 		if (spa_feature_is_active(spa, SPA_FEATURE_LZ4_COMPRESS))
 			result = ZIO_COMPRESS_LZ4_ON_VALUE;
+		else if (spa_feature_is_active(spa, SPA_FEATURE_ZSTD_COMPRESS))
+			result = ZIO_COMPRESS_ZSTD_ON_VALUE;
```

Comment:
This unexpectedly changes the default for 'compress=on' to ZSTD if the feature flag is active. This is likely not intended.

Reply:
only if the feature LZ4 is not available. so the first "if" is always true in fact. so yes. this codepiece can be removed since its dead code

Reply:
Ahh yes, I didn't notice that, ok